src/tests/src/libs/antares/study/thermal-price-definition/thermal-price-definition.cpp

```diff
@@ -0,0 +1,239 @@
+#define BOOST_TEST_MODULE "test thermal price definition"
+
+#define WIN32_LEAN_AND_MEAN
+
+#include <boost/test/included/unit_test.hpp>
+#include <yuni/io/file.h>
+#include <filesystem>
+#include <fstream>
+
+#include <study.h>
+#include <antares/exception/LoadingError.hpp>
+
+#include "../checks/checkApplication.h"
+#include "cluster_list.h"
+#include "container.h"
+
+#define SEP Yuni::IO::Separator
+
+using namespace std;
+using namespace Antares::Data;
+
+std::string getFolder()
+{
+    std::filesystem::path tmpDir = std::filesystem::temp_directory_path();
+    return tmpDir.string();
+}
+
+void createIniFile()
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "list.ini", std::ofstream::out | std::ofstream::trunc);
+
+    outfile << "[area]" << std::endl;
+    outfile << "name = area" << std::endl;
+    outfile << "group = Lignite" << std::endl;
+    outfile << "unitcount = 4" << std::endl;
+    outfile << "nominalcapacity = 900.000000" << std::endl;
+    outfile << "min-stable-power = 400.000000" << std::endl;
+    outfile << "min-up-time = 24" << std::endl;
+    outfile << "min-down-time = 24" << std::endl;
+    outfile << "co2 = 1.200000" << std::endl;
+    outfile << "marginal-cost = 35.000000" << std::endl;
+    outfile << "fixed-cost = 1700.000000" << std::endl;
+    outfile << "startup-cost = 70000.000000" << std::endl;
+    outfile << "market-bid-cost = 35.000000" << std::endl;
+    outfile << "costgeneration = useCostTimeseries" << std::endl;
+    outfile << "efficiency = 35.350000" << std::endl;
+    outfile << "variableomcost = 12.120000" << std::endl;
+
+    outfile.close();
+}
+
+void createFuelCostFile(int size)
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "fuelCost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void createCo2CostFile(int size)
+{
+    std::string folder = getFolder();
+
+    std::ofstream outfile;
+    outfile.open(folder + SEP + "CO2Cost.txt", std::ofstream::out | std::ofstream::trunc);
+
+    for (int i = 0; i < size; i++)
+    {
+        outfile << 1 << std::endl;
+    }
+}
+
+void removeIniFile()
+{
+    std::string folder = getFolder();
+    std::filesystem::remove(folder + SEP + "list.ini");
+}
+
+void removeCostFiles()
+{
+    std::string folder = getFolder();
+    std::filesystem::remove(folder + SEP + "fuelCost.txt");
```

Comment:
Instead of using Yuni::IOSeparator you can use std::filesystempath opetaror "/"
`folder / "fuelCost.txt"` will create a valid path for the plateforrm given "folder" is a filesystempath.